### ISSUE TITLE
Workbox responses with cached data for network only requests

### ISSUE THREAD

**Issue by omerharuncetin** [vite-pwa/sveltekit#79]
Hello, 

I fetch data using a npm package. In that package it makes requests to a graphql endpoint. When I update my profile data and refetch profile information workbox responses with old response instead of the new one. I defined the url pattern in `navigateFallbackDenylist` and added configuration to `runtimeCaching` to prevent caching. What could be the problem?

Here are the details related to config and regex:

regex: /^https:\/\/api-v2-mumbai-live.lens.dev\/.*/

configuration: 
```ts
 {
            urlPattern: /^https:\/\/api-v2-mumbai-live.lens.dev\/.*/,
            handler: "NetworkOnly",
            method: "POST",
            options: {
              matchOptions: {
                ignoreVary: true,
                ignoreSearch: true,
              },
              plugins: [
                {
                  /* this callback will be called when the fetch call fails */
                  handlerDidError: async () =>
                    Response.redirect("/error?offline", 302),
                  /* this callback will prevent caching the response */
                  cacheWillUpdate: async () => null,
                },
              ],
            },
          }
```

**Comment by omerharuncetin** [comment]
I don't want

> You need to navigate to it, then close the browser and repoen it, then navigate again: the workbox cannot create the cache on first request.
> 
> To solve the problem, you need to use a custom sw and warm/create the cache on install event, then it should work: you will need to move all the logic to the custom sw.

I don't want to cache it at all. It should only return the network response. Do I need to create custom service worker still?

**Comment by userquin** [comment]
Oops, sorry, I was thinking about another Nuxt issue...

**Comment by userquin** [comment]
IIRC any request done to another origin shouldn't be intercepted by the sw, you don't need to include any runtime caching: you should handle fetch errors from the app

**Comment by omerharuncetin** [comment]
> Oops, sorry, I was thinking about another Nuxt issue...

That's fine, being super busy can cause these kind of things 😄

**Comment by omerharuncetin** [comment]
> IIRC any request done to another origin shouldn't be intercepted by the sw, you don't need to include any runtime caching: you should handle fetch errors from the app

Yes, I handle the errors in the app however the problem is here that it returns the cached response. I tried to add the config that I provided in topic however it still returns the old response. What could be the problem?

**Comment by userquin** [comment]
You can add `mode: 'development'` to check workbox logs in the sw (in your local if you can check against the external api), check also if the responses from the backend are cached, any api call shouldn't be cached (200 with cache headers like: no-cache no store max-age: 0...)

**Comment by userquin** [comment]
can you show the response headers for graphql requests? maybe you've a bug in your app.

**Comment by omerharuncetin** [comment]
Here is the response header:
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/dfde66c0-a236-44e9-8f33-1a16c2a8074c)

And here is the workbox logs:
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/570cd754-dec9-439c-8e14-cca41711919a)

In addition to that, when I check the `Cache storage` I don't see any value related to those requests.

Also for the request headers, I set `Cache` to `no-cache`

**Comment by userquin** [comment]
check if the browsing is caching the response in network tab; where is the status (should be 200, you have Date and etag response headers, this is wrong, the browser can cache the response if match the criteria => date and etag not changed for example)?

**Comment by userquin** [comment]
Check also if `if-modified-since` present  in the request headers.

**Comment by omerharuncetin** [comment]
first response headers:
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/dd8aab12-906b-4765-9231-b7b7b432b1c8)

second response headers:
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/e847437f-7eb4-497d-8373-5c09d3cc998a)

Also here is the how network tab looks like for that request:
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/6d758117-11a1-4101-a98c-98ededab501a)


No, there is no `if-modified-since` specified in request headers.

**Comment by userquin** [comment]
How about the content?

My suggestion (I use these headers): remove etag adding `Cache-Control: no-cache, no-store, max-age=0, must-revalidate` and `Expires: 0` headers.

**Comment by omerharuncetin** [comment]
> How about the content?
> 
> My suggestion (I use these headers): remove etag adding `Cache-Control: no-cache, no-store, max-age=0, must-revalidate` and `Expires: 0` headers.

By content, if you mean the response data it still returns the old response.

I added headers you mentioned
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/f57f0a5e-d088-4716-807f-a437a3c5e448)

However it is still the same, also the network requests are slightly changed. There is no workbox initiator now.
![image](https://github.com/vite-pwa/sveltekit/assets/47770235/8e825b34-b002-4c64-94d5-ba3803770440)

**Comment by userquin** [comment]
Just remove the runtime caching for external api.

If the response data is old one, you should have something wrong in your graphql backend or proxy: disabling the PWA works?

**Comment by omerharuncetin** [comment]
I removed the runtime caching and tried, nothing changed it still returns the old data.

I disabled the pwa, unregistered the service worker, cleared the site data and tried, nothing changed again.

I don't maintain the graphql api. So, I don't know how they implemented things there and I don't have access to change. However, we tried many things here and also I am struggling with this problem for almost 1 week. Do you think any other solution could work or should I contact with the maintainers of the backend?

**Comment by userquin** [comment]
if using apollo client maybe you can use this default options in the constructor:
```ts
const defaultOptions: DefaultOptions = {
      watchQuery: {
        fetchPolicy: 'no-cache',
        errorPolicy: 'ignore',
      },
      query: {
        fetchPolicy: 'no-cache',
        errorPolicy: 'all',
      },
    }
```

**Comment by omerharuncetin** [comment]
I don't use apollo client. I make the requests by using a sdk. As far as I know they don't use apollo too, only graphql and fetch. And I already set `cache` request header as `no-cache`

**Comment by userquin** [comment]
check if the sdk has some internal flag, check also if the new credentials are sent properly
